infrastructure/bicep/btc-node.bicep

```diff
@@ -0,0 +1,140 @@
+// --- PARAMETERS --- //
+param location string
+
+param pipName string
+param vmName string
+param vmDiskName string
+param nsgName string
+param nicName string
+
+param vmUser string
+@secure()
+param vmPassword string
+
+param subnetId string
+
+param allowRpc bool
+param allowedIpRange string
+
+
+// --- RESOURCES --- //
+resource pip 'Microsoft.Network/publicIPAddresses@2020-11-01' = {
+  name: pipName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+    dnsSettings: {
+      domainNameLabel: vmName
+    }
+  }
+}
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2020-11-01' = {
+  name: nsgName
+  location: location
+  properties: {
+    securityRules: [
+      {
+        name: 'SSH'
+        properties: {
+          protocol: 'TCP'
+          sourcePortRange: '*'
+          destinationPortRange: '22'
+          sourceAddressPrefix: allowedIpRange
+          destinationAddressPrefix: '*'
+          access: 'Allow'
+          priority: 300
+          direction: 'Inbound'
+        }
+      }
+    ]
+  }
+}
+
+resource rpcRule 'Microsoft.Network/networkSecurityGroups/securityRules@2020-11-01' = if (allowRpc) {
+  parent: nsg
+  name: 'RPC'
+  properties: {
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '8332'
+    sourceAddressPrefix: allowedIpRange
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 350
+    direction: 'Inbound'
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-11-01' = {
+  name: nicName
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          publicIPAddress: {
+            id: pip.id
+          }
+          subnet: {
+            id: subnetId
+          }
+        }
+      }
+    ]
+    networkSecurityGroup: {
+      id: nsg.id
+    }
+  }
+}
+
+resource vm 'Microsoft.Compute/virtualMachines@2022-03-01' = {
+  name: vmName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: 'Standard_B2s'
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'canonical'
+        offer: '0001-com-ubuntu-server-focal'
+        sku: '20_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        name: vmDiskName
+        createOption: 'FromImage'
+        caching: 'ReadWrite'
+        managedDisk: {
+          storageAccountType: 'StandardSSD_LRS'
+        }
+        diskSizeGB: 1023
```

Comment:
shouldn't this be a value of power of 2? Or is there a special reason to have it at 2 ^ 10 - 1?

Reply:
![image](https://user-images.githubusercontent.com/85513542/190093361-47cab08f-8827-4c1b-9d8a-f84f245920c9.png)
Don't ask me why 🤔